cmd/experimental/kjobctl/pkg/cmd/describe/describe.go

```diff
@@ -0,0 +1,249 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package describe
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	"k8s.io/cli-runtime/pkg/resource"
+
+	"sigs.k8s.io/kueue/cmd/experimental/kjobctl/pkg/cmd/util"
+	"sigs.k8s.io/kueue/cmd/experimental/kjobctl/pkg/constants"
+)
+
+const (
+	jobExample = `  # Describe a task with job mode
+  kjobctl describe job sample-job
+
+  # Describe a task with job mode
+  kjobctl describe job/sample-job
+  
+  # Describe all tasks with job mode
+  kjobctl describe job
+  
+  # Describe tasks by label name=myLabel
+  kjobctl describe -l name=myLabel`
+)
+
+const (
+	modeTaskArgsFormat = iota
+	modeSlashTaskArgsFormat
+	modeArgsFormat
+)
+
+type DescribeOptions struct {
+	AllNamespaces  bool
+	Namespace      string
+	AppProfileMode string
+	TaskName       string
+	LabelSelector  string
+
+	UserSpecifiedTask []string
+
+	argsFormat      int
+	ResourceGVK     schema.GroupVersionKind
+	ResourceBuilder *resource.Builder
+
+	genericiooptions.IOStreams
+}
+
+func NewDescribeOptions(streams genericiooptions.IOStreams) *DescribeOptions {
+	return &DescribeOptions{
+		IOStreams: streams,
+	}
+}
+
+func NewDescribeCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewDescribeOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:                   "describe APP_PROFILE_MODE NAME",
+		DisableFlagsInUseLine: true,
+		Short:                 "Show details of a specific resource or group of resources.",
+		Example:               jobExample,
+		Args:                  cobra.MatchAll(cobra.RangeArgs(1, 2), cobra.OnlyValidArgs),
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter, args))
+			cobra.CheckErr(o.Run(cmd.Context()))
+		},
```

Comment:
Don't use `cobra.CheckErr` it make the code hard to test (exits the test process before reporting the status).

You can use something like:
```suggestion
RunE: func(cmd *cobra.Command, args []string) error {
trasc marked this conversation as resolved.
			// stop the usage if we've got this far
			cmd.SilenceUsage = true

			err := o.Complete(clientGetter, args)
			if err != nil {
				return err
			}
			return o.Run(cmd.Context())
		},
```